src/QuadForm/Herm/Genus.jl

```diff
@@ -1806,47 +1807,59 @@ function genus_generators(L::HermLat)
     cocycle = Matrix{elem_type(W)}(undef, ll, ll)
     for i in 1:ll
       for j in 1:ll
-        cocycle = zero(W)
+        cocycle[i,j] = zero(W)
       end
     end
     C = collect(Q0)
     ideals = ideal_type(Rabs)[ q00(C[i]) for i in 1:length(C) ]
     for i in 1:ll
-      for j in 1:ll
-        ij = findfirst(isequal(C[i] * C[j]), C)
-        I = ideals[i] * ideals[j] * inv(ideals[ij])
+      for j in 1:i
+        ij = findfirst(isequal(C[i] + C[j]), C)
+        Iabs = ideals[i] * ideals[j] * inv(ideals[ij])
+	      I = EabstoE(Iabs)
         J = I * inv(a(I))
-        ok, x = isprincipal(J)
-        u = f(nnorm(-(ff\FacElem(nf(RR)(norm(x))))))
+	Jabs = EabstoE\J
+        ok, x = isprincipal(Jabs)
+        u = f(nnorm\(-(ff\FacElem(nf(RR)(norm(x))))))
         x = x * u
         @assert norm(x) == 1
-        y = w(V(Int[ valuation(x - 1, PP[i]) >= VD[i] ? F(0) : F(1) for i in 1:length(PP)]))
-        cocycle[i, j] = y
+	if evaluate(x) == 1
```

Comment:
indentation does not seem to match

Reply:
Yes that is weird, on the .jl file I have modified the indentations appear correctly, on the computer

Reply:
There are tabs. Don't use tabs :)